### DISCUSSION TITLE
What is the best way to dynamically load environment variables?

### DISCUSSION THREAD

**Discussion by andrewdleach** [nuxt/nuxt#16776] [Questions]
I recently discovered [Doppler](https://www.doppler.com/) as a method to manage secrets. 

I'm following the [documentation](https://docs.doppler.com/docs/sdk-javascript) describing the recommended approach for fetching secrets from doppler (the Javascript SDK does not yet exist).

I'm able to successfully fetch the secrets -- yay 🎉 -- but now I'm kind of stuck on the best approach for dynamically loading (and accessing) these secrets as env variables.

I read up on the documentation and observed that the best strategy for setting env variables is to use the `runtimeConfig` in the `nuxt.config.[ts|js]` file. 

Is there a best practice for dynamically loading the `runtimeConfig`?

Here's pseudocode of what i've tried so far:

```
export async function loadSecrets() {
  const secrets = await getSecrets();
  process.env.ENV_VAR = secrets['ENV_VAR']
}

loadSecrets();

export default defineNuxtConfig({
  ... 
  runtimeConfig: {
    OTHER_ENV_VARIABLE: 'other_env_variable_value'
    ENV_VAR: process.env.ENV_VAR
  },
```

I did read through the source code and the documentation implies that `NUXT_` will get appended to any variable in `runtimeConfig`, but I'm not able to see either of these variables when I log out to the console during a `setup()` function

Any help in understanding what to do next would be greatly appreciated

**Comment by sevillaarvin** [comment]
I think you have it backwards, you can set your environment variables to NUXT_XXX so that nuxt will be able to automatically detect them and update the runtime config: https://v3.nuxtjs.org/guide/going-further/runtime-config#environment-variables
